README.md

```diff
@@ -32,48 +32,23 @@ Objects in the container are identified by their `reflect.Type` and **everything
 is treated as a singleton**, meaning there can be only one object in the graph
 of a given type.
 
-For more advanced use cases, consider using a factory pattern. That is to say,
+For more advanced use cases, consider using a factory pattern. That is,
 have one object shared as a singleton, capable of creating many instances
 of the same type on demand.
 
 ## Provide
 
-`Provide` adds an object, or a constructor of an object, to the container.
+The `Provide` method adds an object, or a constructor of an object, to the container.
 
-There are two ways to Provide an object:
+There are several ways to `Provide` an object:
 
 1. Provide a pointer to an existing object
-1. Provide a slice, map, array
-1. Provide a "constructor function" that returns one pointer (or interface)
+1. Provide a slice, map, or array
+1. Provide a constructor function that returns one pointer (or interface)
 
-### Provide a Constructor
-
-Constructor is defined as a function that returns one pointer (or
-interface), an optional error, and takes 0-N number of arguments.
+### Provide an object
 
-Each one of the arguments is automatically registered as a **dependency**
-and must also be an interface or a pointer.
-
-```go
-type Type1 struct {}
-type Type2 struct {}
-type Type3 struct {}
-
-c := dig.New()
-err := c.Provide(func(*Type1, *Type2) *Type3 {
-	// operate on Type1 and Type2 to make Type3 and return
-})
-// dig container is now able to provide *Type3 to any constructor.
-// However, note that in the current examples *Type1 and *Type2
-// have not been provided. Constructors (or instances) first
-// have to be provided to the dig container before it is able
-// to create a shared singleton instance of *Type3
-```
-
-### Provide an Object
-
-Registering an object directly is a shortcut to register something that
-has no dependencies.
+As a shortcut for objects without dependencies, register it directly.
```

Comment:
I think it's weird to talk about "a shortcut" before explaining what the full version is